crates/project/src/lsp_command.rs

```diff
@@ -3020,3 +3028,179 @@ impl LspCommand for LinkedEditingRange {
         BufferId::new(message.buffer_id)
     }
 }
+
+fn lsp_diagnostic_to_diagnostic(diagnostic: lsp::Diagnostic) -> Diagnostic {
```

Comment:
I am thinking about moving this to `language.rs`

Reply:
Pull-based diagnostics go through `LspStore` methods to fill these properties correctly now.